src/sync/heartbeat.ts

```diff
@@ -0,0 +1,40 @@
+import type {ClientID} from './client-id';
+import type * as dag from '../dag/mod';
+import {getClient, setClient} from './clients';
+
+const HEARTBEAT_INTERVAL_MS = 60 * 1000;
+
+export function startHeartbeats(
+  clientID: ClientID,
+  dagStore: dag.Store,
+): () => void {
+  const intervalID = window.setInterval(async () => {
+    await dagStore.withWrite(async (write: dag.Write) => {
+      await writeHeartbeat(clientID, write);
+      await write.commit();
+    });
+  }, HEARTBEAT_INTERVAL_MS);
+  return () => {
+    window.clearInterval(intervalID);
+  };
+}
+
+export async function writeHeartbeat(
+  clientID: ClientID,
+  write: dag.Write,
+): Promise<void> {
+  const client = await getClient(clientID, write);
+  if (!client) {
+    // Should this be a more specific error so caller can detect and handle?
+    throw new Error('Cannot write heartbeat. Client with clientID not found');
+  }
+
+  await setClient(
+    clientID,
+    {
+      ...client,
```

Comment:
Can you mark the Client type as readonly so we do not accidentally mutate it?

Reply:
I've been careful to always write these JSON objects in the same order as described in the typedef. The order matters when computing the hash. Things will not break but generally we want to ensure that we end up with the same hash for 2 json values that are deep equal.

In this specific case it is not going to matter because `heartbeatTimestampMs` is changing every time.

Reply:
done.  

the object construction details impacting hash is a nasty gotcha.

Reply:
Yeah! I wonder if we can put a lint pass in that they are alpha?

Reply:
(Note: hash mismatches don’t affect our correctness, only performance)